README.md

```diff
@@ -207,31 +162,136 @@ Users want to determine several types of the formatting width as following
+  * `style: "long" | "short" | "narrow" | "digital"`: The base style to be used for formatting. This can be overriden per-unit by setting the more granular options. Defaults to `"short"`.
+  * `years: "long" | "short" | "narrow"`: The style to be used for formatting years.
+  * `yearsDisplay: "always" | "auto"`: Whether to always display years, or only if nonzero.
+  * `months: "long" | "short" | "narrow"`: The style to be used for formatting months.
+  * `monthsDisplay: "always" | "auto"`: Whether to always display months, or only if nonzero.
+  * `weeks: "long" | "short" | "narrow"`: The style to be used for formatting weeks.
+  * `weeksDisplay: "always" | "auto"`: Whether to always display weeks, or only if nonzero.
+  * `days: "long" | "short" | "narrow"`: The style to be used for formatting days.
+  * `daysDisplay: "always" | "auto"`: Whether to always display days, or only if nonzero.
+  * `hours: "long" | "short" | "narrow" | "numeric" | "2-digit"`: The style to be used for formatting hours.
+  * `hoursDisplay: "always" | "auto"`: Whether to always display hours, or only if nonzero.
+  * `minutes: "long" | "short" | "narrow" | "numeric" | "2-digit"`: The style to be used for formatting minutes.
+  * `minutesDisplay: "always" | "auto"`: Whether to always display minutes, or only if nonzero.
+  * `seconds: "long" | "short" | "narrow" | "numeric" | "2-digit"`: The style to be used for formatting seconds.
+  * `secondsDisplay: "always" | "auto"`: Whether to always display seconds, or only if nonzero.
+  * `milliseconds: "long" | "short" | "narrow" | "numeric"`: The style to be used for formatting milliseconds.
+  * `millisecondsDisplay: "always" | "auto"`: Whether to always display milliseconds, or only if nonzero.
+  * `microseconds: "long" | "short" | "narrow" | "numeric"`: The style to be used for formatting microseconds.
+  * `microsecondsDisplay: "always" | "auto"`: Whether to always display microseconds, or only if nonzero.
+  * `nanoseconds: "long" | "short" | "narrow" | "numeric"`: The style to be used for formatting nanoseconds.
+  * `nanosecondsDisplay: "always" | "auto"`: Whether to always display nanoseconds, or only if nonzero.
+  * `fractionalDigits: number`: The number of fractional digits to be included in the output, applies automatically to the entire set of consecutive time units with style `"numeric"` at the end.
+
+##### Default values
+
+* The per-unit style options default to the value of `style` for all styles except `"digital"`, for which units years till days default to `"narrow"` and hours till nanoseconds default to `"numeric"`.
+* The per-unit display options default to `"auto"` if the corresponding style option is `undefined` and `"always"` otherwise.
+
+#### Notes
+
+* If for any given locale and unit, the representation of the `"narrow"` style is completely unique, then the same repsresentation may be used for `"short"`.
```

Comment:
Was below what you had in mind?  If so I'd suggest clarifying, as I wasn't quite sure what the line above meant.

```suggestion
* Some locales may share the same representation between `"long"` and `"short"` or between `"narrow"` and `"short"`.  Others may use different representations for each one, e.g. "3 seconds", "3 secs", "3s".
```

Reply:
This looks great, merging.